.github/workflows/main.yml

```diff
@@ -2,12 +2,11 @@ name: test
 
 on:
   push:
-    branches: [main]
   pull_request:
-    branches: [main]
   schedule:
```

Comment:
Let's keep `branches: [main]` from `push`? Also, I was a bit surprised that this was changed in a commit bumping the actions' versions given the granularity of other commits. :)

With this change, any branch pushed to this project will trigger a CI run; which could result in duplicated CI runs for something like a README typo fix submitted by a maintainer.

Reply:
Unfortunately with `push: branches: [main]`, it's not possible to test on our forks before opening PRs, and I wouldn't have contributed this or #259 in the first place.

Looking at recent PRs, all maintainers create PRs from their own forks. For the occasional typo fix made directly in this repo, I think the duplicate build is fine, and it's much more important to test forks.

If it's not fine, how does this config look?

```yml
    # We want to run on external PRs, but not on our own internal PRs as they'll be run
    # by the push to the branch.
    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
```

re: https://github.com/has2k1/plotnine/blob/f696c01f1602e422a29778c112005463992ed707/.github/workflows/testing.yml#L10-L12

Reply:
Ah, good point. I guess leaving this as-is works fine; since the cost is a duplicated run which will fail the same was as other runs.